### PR TITLE
add stats for coverage query starts

### PR DESCRIPTION
**Pull request by evanmcc**
- index fsm: create, create_error, active
- keys and bucket fsms (as 'list'): create, create_error, active

This should give us a better picture as to how many index and list queries are being executed against each riak node.

cc @russelldb

### CODE REVIEW

File: src/riak_kv_stat.erl [hunk 1]
```diff
@@ -102,12 +104,28 @@ stop() ->
 
 init([]) ->
     register_stats(),
-    {ok, ok}.
+    State = {state, [spawn(?MODULE, monitor_loop, [index]),
```
Comment (evanmcc) [inline src/riak_kv_stat.erl]: should these be spawn_link() instead?

File: src/riak_kv_stat_bc.erl [hunk 2]
```diff
@@ -232,7 +232,13 @@ legacy_stat_map() ->
      {coord_redirs_total, {riak_kv,node,puts,coord_redirs}, counter},
      {executing_mappers, {riak_kv,mapper_count}, counter},
      {precommit_fail, {riak_kv, precommit_fail}, counter},
-     {postcommit_fail, {riak_kv, postcommit_fail}, counter}
+     {postcommit_fail, {riak_kv, postcommit_fail}, counter},
+     {index_fsm_create, {{riak_kv, index, fsm, create}, one}, spiral},
+     {index_fsm_create_error, {{riak_kv, index, fsm, create}, one}, spiral},
+     {index_fsm_active, {riak_kv, index, fsm, active}, counter},
+     {list_fsm_create, {{riak_kv, list, fsm, create}, one}, spiral},
+     {list_fsm_create_error, {{riak_kv, list, fsm, create}, one}, spiral},
```
Comment (russelldb) [inline src/riak_kv_stat_bc.erl]: Should be `{riak_kv, list, fsm, create, error}`

File: src/riak_kv_stat_bc.erl [hunk 3]
```diff
@@ -232,7 +232,13 @@ legacy_stat_map() ->
      {coord_redirs_total, {riak_kv,node,puts,coord_redirs}, counter},
      {executing_mappers, {riak_kv,mapper_count}, counter},
      {precommit_fail, {riak_kv, precommit_fail}, counter},
-     {postcommit_fail, {riak_kv, postcommit_fail}, counter}
+     {postcommit_fail, {riak_kv, postcommit_fail}, counter},
+     {index_fsm_create, {{riak_kv, index, fsm, create}, one}, spiral},
+     {index_fsm_create_error, {{riak_kv, index, fsm, create}, one}, spiral},
```
Comment (russelldb) [inline src/riak_kv_stat_bc.erl]: Should be `{riak_kv, index, fsm, create, error}`

**Comment by russelldb** [comment]
Running each of bucket list, key list of index query and then running stats shows incorrect results. The stats output shows that for each [index | list]_fsm_create there is an [index | list]_fsm_create_error. I've highlighted the lines above that are the cause of this (incorrect stat used in error output.) Probably just copypasta.

That aside, it looks good to me, and it works (checking folsom shows correct stats updated.)

File: src/riak_kv_stat_bc.erl (hunk 2 above)
Comment (evanmcc) [inline src/riak_kv_stat_bc.erl]: oops

**Comment by evanmcc** [comment]
updated.

**Comment by russelldb** [comment]
wfm +1 to merge. Many thanks!

**Pull request merged**